iceoryx_posh/include/iceoryx_posh/internal/roudi/port_manager.inl

```diff
@@ -16,20 +17,27 @@
 #ifndef IOX_POSH_INTERNAL_ROUDI_PORT_MANAGER_INL
 #define IOX_POSH_INTERNAL_ROUDI_PORT_MANAGER_INL
 
+#include "iceoryx_posh/internal/roudi/port_manager.hpp"
+
 namespace iox
 {
 namespace roudi
 {
 template <typename T, std::enable_if_t<std::is_same<T, iox::build::OneToManyPolicy>::value>*>
 inline cxx::optional<RuntimeName_t>
-PortManager::doesViolateCommunicationPolicy(const capro::ServiceDescription& service) const noexcept
+PortManager::doesViolateCommunicationPolicy(const capro::ServiceDescription& service) noexcept
 {
     // check if the publisher is already in the list
     for (auto publisherPortData : m_portPool->getPublisherPortDataList())
     {
         popo::PublisherPortRouDi publisherPort(publisherPortData);
         if (service == publisherPort.getCaProServiceDescription())
         {
+            if (publisherPortData->m_toBeDestroyed)
+            {
+                destroyPublisherPort(publisherPortData);
+                continue;
```

Comment:
Ok, we ignore those that are about to be destroyed in the violation check (i.e. whether there already is such a port).

Now it might be possible that a port is about to be marked for destruction but not quite (i.e. `m_toBeDestroyed == false`) but we deny another port to be constructed. This is ok I guess, since we could argue that at this point the port still lives (and hence we cannot construct another one).

If so, the logic is fine.